### DISCUSSION TITLE
Running tests on a target

### DISCUSSION THREAD

**Discussion by Kasslim** [micropython/micropython#9101] [Core Development] [answered]
Background: I'm running tests on an stm32 target, seeing many failures and crashes in some `tests/` subfolders.
Example: I found `cpydiff/` to be a collection of failing tests meant to show the diff between Micropython and standard Python. 

Besides `basics/`, which subfolders/tests should I expect to pass on a target? I haven't found any README or docs page with descriptions of all subfolders so any hints are welcome.

**Comment by jimmo** [comment, marked as answer]
> Example: I found cpydiff/ to be a collection of failing tests meant to show the diff between Micropython and standard Python.

Yes the cpydiff is designed to be run as part of the docs generation. It uses a different script (tools/gen-cpydiff.py) to run these tests.

> Besides basics/, which subfolders/tests should I expect to pass on a target? I haven't found any README or docs page with descriptions of all subfolders so any hints are welcome.

I would expect the following directories to all pass on an stm32 target.

`basics`, `micropython`, `misc`, `extmod`, `float`, `stress`, `pyb`, `inlineasm`

This is actually what run-tests.py does when it isn't given any directories -- it figures out based on the `--target` argument and will use those directories for a pyboard. You can see the logic to choose the appropriate directories here:
https://github.com/micropython/micropython/blob/master/tests/run-tests.py#L943

A historical note -- the stm32 port was originally written for the pyboard, hence why some things still use "pyb" to refer to "stm32".
